File: week-2/week2-DC/q3.cpp
#include<iostream>
using namespace std;
int main(){
    int first_term,common_ratio,n,k;     //n=no. of terms of series
                                         //k=no. of terms to sum  
    cout<<"Enter the first term:";
    cin>>first_term;

    cout<<"Enter the common ratio:";
    cin>>common_ratio;

    cout<<"Enter the number of terms to be printed:";
    cin>>n;

    cout<<"Enter the number of terms to be summed up:";
    cin>>k;

    int c[n],sum=0;
    c[0]=first_term;

    for(int i=1;i<n;i++){
        int a=1,b;
        c[i]=c[i-1]*common_ratio;
    }
    cout<<"GP:";
    for(int i=0;i<n;i++){
        cout<<c[i]<<" ";
    }
    for(int i=0;i<k;i++){
        sum=sum+c[i];
    }
    cout<<"\nSum upto "<<k<<"th term:"<<sum<<endl;

    return 0;
}

File: week-2/week2-DC/q3.py
first_term=int(input("Enter the first term:"))
common_ratio=int(input("Enter the common ratio:"))
n=int(input("Enter the number of terms to be printed:"))#n=no.of terms of series
k=int(input("Enter the number of terms to be summed up:"))#k=no.of terms to sum
a=first_term
r=common_ratio
value=a
gp=[a]
print("GP:",a)
for i in range(n-1):
    value*=r
    print(value)
    gp.append(value)
sum=0
for i in range(k):
    sum+=gp[i]
print("Sum upto",k,"th term:",sum)    